internal/resources/kv/kv.go

```diff
@@ -0,0 +1,105 @@
+package kv
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/ydb-platform/ydb-go-genproto/draft/protos/Ydb_KeyValue"
+
+	"github.com/ydb-platform/terraform-provider-ydb/internal/helpers"
+)
+
+type MediaConfig struct {
+	Media string
+}
+
+type ChannelConfig struct {
+	Channel []*MediaConfig
+}
+
+type Resource struct {
+	Entity *helpers.YDBEntity
+
+	UseTLS           bool
+	FullPath         string
+	Endpoint         string
+	Database         string
+	Path             string
+	DatabaseEndpoint string
+	PartitionCount   int
+	StorageConfig    *ChannelConfig
+}
```

Comment:
Here where structcheck linter can help. For better memory allocation(because of alignment) you can do something like

```
struct {
Entity *helpers.Entity
StorageConfig *ChannelConfig
	FullPath         string
	Endpoint         string
	Database         string
	Path             string
	DatabaseEndpoint string
	PartitionCount   int
UseTLS bool
```

UseTLS should be the last variable, because of alignment. `bool` is only one byte size, so if some 8 byte type comes next (string, int) we use 16 bytes instead of 9 :)

Reply:
fixed